### PR TITLE
fix some bugs of Paranoid Pirate

### PR DESCRIPTION
**Pull request by TianPeilong**
1、Add timer to poller；
2、Fix the bug of  the purge method;
3、Fix the problem of expiry calculation of worker.

### CODE REVIEW

File: src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ private static void Main(string[] args)
             // if liveliness == 0 -> queue is considered dead/disconnected
             var liveliness = Commons.HeartbeatLiveliness;
             var interval = Commons.IntervalInit;
-            var heartbeatAt = DateTime.UtcNow.Millisecond + Commons.HeartbeatInterval;
+            var heartbeatAt = DateTime.UtcNow + + TimeSpan.FromMilliseconds(Commons.HeartbeatInterval);
```
Comment (drewnoakes) [inline src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs]: Double `+ +` here?

File: src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs [hunk 2]
```diff
@@ -116,9 +116,9 @@ private static void Main(string[] args)
                     }
                 }
                 // if it is time the worker will send a heartbeat so QUEUE can detect a dead worker
-                if (DateTime.UtcNow.Millisecond > heartbeatAt)
+                if (DateTime.UtcNow > heartbeatAt)
                 {
-                    heartbeatAt = DateTime.UtcNow.Millisecond + Commons.HeartbeatInterval;
+                    heartbeatAt = DateTime.UtcNow + + TimeSpan.FromMilliseconds(Commons.HeartbeatInterval);
```
Comment (drewnoakes) [inline src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs]: Ditto.

**Review by drewnoakes — COMMENTED** [review]
Looks good apart from the two highlighted problems.

File: src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs (hunk 1 above)
Comment (TianPeilong) [reply src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs]: sorry, I have fix it.

File: src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs (hunk 2 above)
Comment (TianPeilong) [reply src/Pirate Pattern/Paranoid Pirate/ParanoidPirate.Worker/Program.cs]: Same.

**Comment by drewnoakes** [comment]
@TianPeilong thank you very much.

**Pull request merged**